### PR TITLE
Base repository setup.

### PR DESCRIPTION
**Pull request by bdice**
This PR sets up some basic configuration for the repository. I am setting it up as a Python project modeled after https://github.com/rapidsai/dependency-file-generator. I left out the "semantic release" configuration from that repository for now, since I don't totally understand it, but I may solicit help from @ajschmidt8 with that in the future.

We won't be releasing this to PyPI -- this repository will be used as the source, and git tags will act as the version indicators used downstream in `.pre-commit-config.yaml` files. For that reason, we shouldn't need to track RAPIDS versions or branches like `branch-YY.MM`. This repo should be versioned independently, since it will always be pinned at a specific tag by downstream projects.

### CODE REVIEW

**Comment by bdice** [comment]
After this PR is merged, I will start work on the copyright checker hook. I am thinking we will want to have a CLI application for each hook, similar to https://github.com/pre-commit/pre-commit-hooks. Reviewers, please share your thoughts on that.

**Comment by wence-** [comment]
Just to note if attempting to consolidate a bunch of these checks. It may well be worth assessing [ruff](https://beta.ruff.rs/) as an alternative to flake8/isort/pydocstyle (and probably soon, black).

File: .flake8
Comment (bdice) [reply .flake8:2]: Yes, this is repo-specific config. I copied it straight from the dependency-file-generator repo.

**Comment by bdice** [comment]
> Just to note if attempting to consolidate a bunch of these checks. It may well be worth assessing [ruff](https://beta.ruff.rs/) as an alternative to flake8/isort/pydocstyle (and probably soon, black).

The actual contents of this repo will be limited, mostly to define some common hooks like the copyright checker. We won’t be implementing or configuring any of the existing Python hooks like flake8, isort, etc. here.

**Comment by vyasr** [comment]
Agree with Bradley on limiting this repo. That said, I do plan to investigate ruff for the rest of RAPIDS at some point.

File: .flake8
Comment (vyasr) [reply .flake8:2]: I don't have a strong pref here although I would generally prefer to stick with either the black (88) or flake8 (79) defaults since I have found that other heuristics in linters are occasionally designed in such a way that they produce suboptimal results on very different line lengths. Perhaps that situation has improved though.

**Comment by vyasr** [comment]
@bdice I don't recall why I didn't review this PR when I initially saw it; I was certain I had a reason but I can't remember it any more 😅 should I go ahead?

**Comment by bdice** [comment]
@vyasr Yeah sure! I will probably take one round of feedback and merge this, so it doesn’t hinder the actual hook development I want to do.

File: .flake8
Comment (vyasr) [inline .flake8:3]: Is this just to avoid picking up codes from flake8 plugins if someone happens to install them?

File: .flake8
Comment (vyasr) [inline .flake8:1]: I know we aren't the most consistent across all repos, but given that this is a new repo we should be consistent and add the full copyright notice (not just the one-line header) to all files. That was the most current advice (still waiting on final word, but in the interest of moving this PR forward I would add it everywhere and be done with it).

File: pyproject.toml
Comment (vyasr) [inline pyproject.toml:21]: No need to preemptively add an empty list.
```suggestion
```

File: pyproject.toml
Comment (vyasr) [inline pyproject.toml:9]: Is this a real email that we would respond to? I would rather just leave one out unless it really is a maintained address.

File: src/rapids_pre_commit_hooks/LICENSE
Comment (vyasr) [inline src/rapids_pre_commit_hooks/LICENSE:1]: This symlink isn't necessary AFAICT.

**Review by vyasr — CHANGES_REQUESTED** [review]
Well this is the perfect PR to make use of file-level comments!

File: .flake8
Comment (bdice) [reply .flake8:3]: I copied from `dependency-file-generator`. If you have a preferred source for a default flake8 configuration, I will replace it with whatever you prefer.

You are correct that `select = E,F,W` limits the flake8 codes to the base set, and will not check error codes from plugins.

File: .flake8
Comment (bdice) [reply .flake8:1]: I am happy to copy-paste if you have examples of what this should look like, but I don't want to determine that precedent right now. I would say this is in scope for a later version of the repo, because we can apply an updated copyright checker script (which will be hosted in this repo) to the repo itself.

File: pyproject.toml
Comment (bdice) [reply pyproject.toml:9]: I assume it is real, because it was listed in another package. If it is not real, we should fix that upstream. cc: @ajschmidt8 Can you confirm?

Comment (bdice) [reply pyproject.toml:23]: @raydouglass confirmed that this email would be received and could be set up to forward to whatever list of people needed.

File: .flake8
Comment (vyasr) [reply .flake8:17]: I'm fine leaving this list for now. We can update later.

**Review by vyasr — APPROVED** [review]

**Comment by bdice** [comment]
Just realized I clicked Merge and didn’t do /merge. What should the protocol be here? Should merges be squashed? Should they be gated by the /merge command?

**Comment by vyasr** [comment]
@bdice I don't think we need the ops bot here, but I would like to switch the merge strategy to squashing. @ajschmidt8 could you help with that?

**Comment by ajschmidt8** [comment]
> @bdice I don't think we need the ops bot here, but I would like to switch the merge strategy to squashing. @ajschmidt8 could you help with that?

Done. Squash merging was previously enabled alongside merge commits and rebase merging.

I've just disabled merge commits and rebase merging so that squash commits are the only allowed method.

**Pull request merged**
